raycluster_webhook/src/patches.go

```diff
@@ -0,0 +1,352 @@
+			},
+			VolumeMounts: key_volumes,
+		}
+		// Append WorkerGroup init container
+		patches = append(patches, patchOperation{
+			Op:    "replace",
+			Path:  "/spec/workerGroupSpecs/0/template/spec/initContainers/0",
+			Value: initContainerWorker,
+		})
+
+		workerCaVolume := corev1.Volume{
+			Name: "ca-vol",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: secretName,
+				},
+			},
+		}
+
+		workerServerVolume := corev1.Volume{
+			Name: "server-cert",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		}
+
+		patches = append(patches, patchOperation{
+			Op:   "add",
+			Path: "/spec/workerGroupSpecs/0/template/spec/containers/0/env/-",
+			Value: map[string]interface{}{
+				"name": "MY_POD_IP",
+				"valueFrom": map[string]interface{}{
+					"fieldRef": map[string]interface{}{
+						"fieldPath": "status.podIP",
+					},
+				},
+			},
+		}, patchOperation{
+			Op:    "add",
+			Path:  "/spec/workerGroupSpecs/0/template/spec/containers/0/env/-",
+			Value: map[string]interface{}{"name": "RAY_USE_TLS", "value": "1"},
+		}, patchOperation{
+			Op:    "add",
+			Path:  "/spec/workerGroupSpecs/0/template/spec/containers/0/env/-",
+			Value: map[string]interface{}{"name": "RAY_TLS_SERVER_CERT", "value": "/home/ray/workspace/tls/server.crt"},
+		}, patchOperation{
+			Op:    "add",
+			Path:  "/spec/workerGroupSpecs/0/template/spec/containers/0/env/-",
+			Value: map[string]interface{}{"name": "RAY_TLS_SERVER_KEY", "value": "/home/ray/workspace/tls/server.key"},
+		}, patchOperation{
+			Op:    "add",
+			Path:  "/spec/workerGroupSpecs/0/template/spec/containers/0/env/-",
+			Value: map[string]interface{}{"name": "RAY_TLS_CA_CERT", "value": "/home/ray/workspace/tls/ca.crt"},
+		})
+
+		patches = append(patches, patchOperation{
+			Op:    "add",
+			Path:  "/spec/workerGroupSpecs/0/template/spec/volumes/-",
+			Value: workerCaVolume,
+		})
+
+		patches = append(patches, patchOperation{
+			Op:    "add",
+			Path:  "/spec/workerGroupSpecs/0/template/spec/volumes/-",
+			Value: workerServerVolume,
+		})
+	}
+
+	return patches, nil
+}
+
+func annotationBoolVal(cluster *rayv1api.RayCluster, annotation string, defaultValue bool) bool {
+	val, exists := cluster.ObjectMeta.Annotations[annotation]
+	if !exists || val == "" {
+		return defaultValue
+	}
+	boolVal, err := strconv.ParseBool(val)
+	if err != nil {
+		fmt.Errorf("Could not convert annotation value to bool", "annotation", annotation, "value", val)
+		return defaultValue
+	}
+	return boolVal
+}
+
+func getDomainName() (string, error) {
+	consoleRoute := &routev1.Route{}
+	fmt.Println("1")
+	fmt.Printf(client.ObjectKey{Name: NameConsoleLink, Namespace: NamespaceConsoleLink}.String())
+	fmt.Println("2")
+	fmt.Printf(consoleRoute.String())
+
+	if err := k8Client.Get(context.TODO(), types.NamespacedName{Name: NameConsoleLink, Namespace: NamespaceConsoleLink}, consoleRoute); err != nil {
```

Comment:
But this fails.

So, something is misbehaving into this `.Get()`.

1. It's probably not the `context.TODO()` 
2. I think we fixed the `types.NamespacedName` Struct but we should definitely try to confirm that.
3. The `consoleRoute` seemed reasonable from what I recall as that was the object that was passed to be updated.

We should definitely confirm these 3 things though and I'd dig into them slightly differently.

Reply:
What is the error?
Do you run the container locally, connecting to the remote OpenShift cluster, or directly as a Deployment on the cluster?
If the latter, I see that you doesn't provide RBAC for accessing Routes - https://github.com/project-codeflare/codeflare-operator/pull/508/files#diff-d44acd75643a4ccbb2ddc671e641c9f520d495dbc01fe1dbcf305bd4c13915d9R53-R61

Reply:
@sutaakar This was the error we were receiving. I will update RBAC:
`no kind is registered for the type v1.Route in scheme "go/pkg/mod/k8s.io/apimachinery@v0.29.3/pkg/runtime/scheme.go:100"`